numerical_methods/secant_method.cpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * \file
+ * \brief Solve the equation \f$f(x)=0\f$ using
+ * [Secant Method](https://en.wikipedia.org/wiki/Secant_method)
+ *
+ * The example function \f$f(x)\f$ looks like this:
+ * \f[
+ * x^4 - 29 = 0
+ * \f]
+ *
+ * \author [Fatih Küçükkarakurt](https://github.com/fkkarakurt)
+ */
+
+// Secant Method for the root of f(x)
+// Analytically the solution is x^4-29 = 0
+// => x = 29^(1/4) = 2,3205957871060836757951626724362...
+#include <cmath>
+#include <iomanip>
+#include <iostream>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```